Aplib.Tests/Core/MetadataTests.cs

```diff
@@ -0,0 +1,99 @@
+using Aplib.Core;
+using FluentAssertions;
+
+namespace Aplib.Tests.Core;
+
+public class MetadataTests
+{
+    [Fact]
+    public void Metadata_WhenConstructed_ContainsCorrectMetaData()
+    {
+        // Arrange
+        const string name = "A program without bugs";
+        const string description = "This goal is unreachable \ud83d\ude22";
+
+        // Act
+        Metadata data = new(name, description);
+
+        // Assert
+        data.Should().NotBeNull();
+        data.Id.Should().NotBeEmpty();
```

Comment:
This is a weak test, as `Guid` is a struct and will always not be empty (the default value). As such, it would be better if you check if the guid is not `Guid.Empty`.

Reply:
That is actually what this does :) <https://fluentassertions.com/guids/>

Reply:
Well, TIL. Thanks 😄